app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthAction.scala

```diff
@@ -57,199 +50,66 @@ import scala.util.control.NonFatal
-        logger.error("Error authorising CSP", e)
-        Left(ErrorInternalServerError)
-    }
-  }
-
-  private def maybeBadgeIdentifier[A](implicit vhr: ValidatedHeadersRequest[A]): Option[BadgeIdentifier] = {
+  protected def maybeBadgeIdentifier[A](implicit vhr: ValidatedHeadersRequest[A]): Either[ErrorResponse, BadgeIdentifier] = {
     val maybeBadgeId: Option[String] = vhr.request.headers.toSimpleMap.get(CustomHeaderNames.XBadgeIdentifierHeaderName)
-    maybeBadgeId.filter(xBadgeIdentifierRegex.findFirstIn(_).nonEmpty).map(BadgeIdentifier)
-  }
-
-  // pure function that tames exceptions throw by HMRC auth api into an Either
-  // this enables calling function to not worry about recover blocks
-  // returns a Future of Left(Result) on error or a Right(AuthorisedRequest) on success
-  protected def authoriseAsNonCspNrsEnabled[A](implicit vhr: ValidatedHeadersRequest[A]): Future[Either[ErrorResponse, AuthorisedRequest[A]]] = {
-    implicit def hc(implicit rh: RequestHeader): HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(rh.headers)
-
-    authorised(Enrolment("HMRC-CUS-ORG") and AuthProviders(GovernmentGateway)).retrieve(nonCspRetrievals) {
-      case internalId ~ externalId ~ agentCode ~ credentials ~ confidenceLevel ~ nino ~ saUtr ~ name ~ dateOfBirth ~ email ~ agentInformation ~ groupIdentifier ~
-        credentialRole ~ mdtpInformation ~ itmpName ~ itmpDateOfBirth ~ itmpAddress ~ affinityGroup ~ credentialStrength ~ loginTimes ~ authorisedEnrolments =>
-
-        val maybeEori: Option[Eori] = findEoriInCustomsEnrolment(authorisedEnrolments, hc.authorization)
-
-        logger.debug(s"EORI from Customs enrolment for non-CSP request: $maybeEori")
-        maybeEori.fold[Future[Either[ErrorResponse, AuthorisedRequest[A]]]]{
-          googleAnalyticsConnector.failure(errorResponseEoriNotFoundInCustomsEnrolment.message)
-          Future.successful(Left(errorResponseEoriNotFoundInCustomsEnrolment))
-        }{ eori =>
-          logger.debug("Authorising as non-CSP")
-
-          val retrievalData = NrsRetrievalData(internalId, externalId, agentCode, credentials, confidenceLevel, nino, saUtr,
-            name, dateOfBirth, email, agentInformation, groupIdentifier, credentialRole, mdtpInformation, itmpName,
-            itmpDateOfBirth, itmpAddress, affinityGroup, credentialStrength, loginTimes)
-
-          Future.successful(Right(vhr.toNonCspAuthorisedRequest(eori, Some(retrievalData))))
-        }
-    }.recover{
-      case NonFatal(_: AuthorisationException) =>
-        googleAnalyticsConnector.failure(errorResponseUnauthorisedGeneral.message)
-        Left(errorResponseUnauthorisedGeneral)
-      case NonFatal(e) =>
-        logger.error("Error authorising non-CSP", e)
-        Left(ErrorInternalServerError)
+    maybeBadgeId.filter(xBadgeIdentifierRegex.findFirstIn(_).nonEmpty).map(BadgeIdentifier).toRight[ErrorResponse]{
+      logger.error("badge identifier invalid or not present for CSP")
```

Comment:
Can you use the header name instead? e.g.
`logger.error(s"$XBadgeIdentifierHeaderName invalid or not present for CSP")`